pkg/controller.v1beta2/tensorflow/pod.go

```diff
@@ -119,6 +123,20 @@ func (tc *TFController) reconcilePods(
 				}
 			}
 
+			// According to the issue kubernetes/kubernetes#51333,
+			// k8s would not delete the pod on the node if the node is down.
+			// tf-operator should take care of this case by deleting the pod
+			// so that a new pod will be created and scheduled to healthy node.
+			// if the node is down, the pod's phase is sitll "Running"
+			// and its reason is "NodeLost", so we use reason to check.
+			if pod.Status.Reason == nodeLostReason {
+				logger.Infof("Need to delete pod on lost node: %v.%v", pod.Namespace, pod.Name)
+				err = tc.deletePodNow(pod)
```

Comment:
I am not sure if we just need to delete the pod. In some cases, if one pod is down, the whole job is failed. For example, chief worker.

Reply:
For case that the chief worker is down and restart policy is set to `OnFailure`,   I will test it and see if the fix work.